tests/GenericCollectionTestCase.php

```diff
@@ -5,35 +5,86 @@
 
 abstract class GenericCollectionTestCase extends TestCase
 {
-    protected function assertCollectionReturnsCopyOfArray(GenericCollection $collection, array $originalValues) : void
+    //array of values used to construct the GenericCollection as specified by the subclass
+    protected $values;
+    //GenericCollection instance as specified by the subclass
+    protected $instance;
+    //Concrete type of GenericCollection as specified by the subclass
+    protected $concreteType;
```

Comment:
By using a parameterized type we are able to push most of the tests up into this parent test case.